src/NexusMods.DataModel/FileOriginRegistry.cs

```diff
@@ -60,7 +62,14 @@ public async ValueTask<DownloadId> RegisterDownload(IStreamFactory factory, Meta
 
         await using var tmpFolder = _temporaryFileManager.CreateFolder();
         await _extractor.ExtractAllAsync(factory, tmpFolder.Path, token);
-        return await RegisterFolderInternal(tmpFolder.Path, metaData, null, _fileStore.GetFileHashes(), archiveHash.Value, archiveSize, token);
+
+        return await RegisterFolderInternal(tmpFolder.Path, AppendNestedArchiveMetadata, null, _fileStore.GetFileHashes(), archiveHash.Value, archiveSize, token);
+
+        void AppendNestedArchiveMetadata(ITransaction tx, EntityId id)
+        {
+            metaData?.Invoke(tx, id);
+            tx.Add(id, NestedArchiveMetadata.NestedArchive, new Null());
+        }
```

Comment:
Not sure I like this solution. You're calling this "Nested Archive" because the only usage of this is from trying to use a nested archive, however, the method is for registering a generic stream. It's not guaranteed to come from a nested archive.

You should rename this method to something more representative.

Reply:
Do you have suggestions, I couldn't think of anything so I used that.

Reply:
Call it what it is, a stream-based download. It's not from a file, but from a generic stream.